src/main/java/org/jabref/model/database/BibDatabaseContext.java

```diff
@@ -220,10 +219,12 @@ public static Path getFulltextIndexBasePath() {
 
     public Path getFulltextIndexPath() {
         Path appData = getFulltextIndexBasePath();
-        LOGGER.info("Index path for {} is {}", getDatabasePath().get(), appData.toString());
+
         if (getDatabasePath().isPresent()) {
+            LOGGER.info("Index path for {} is {}", getDatabasePath().get(), appData.toString());
             return appData.resolve(String.valueOf(this.getDatabasePath().get().hashCode()));
         }
+
         return appData.resolve("unsaved");
     }
```

Comment:
I wonder if not that method should return the optional instead of 'unsaved', but this would probably be a bigger refactoring - if it makes any sense...

Reply:
I think it makes a lot of sense. Return an empty optional to represent an invalid path seems a better approach to me.